lib/spack/spack/cmd/edit.py

```diff
@@ -112,9 +117,30 @@ def edit(parser, args):
     if args.path:
         path = args.path
         if name:
-            path = join_path(path, name + ".py")
-            if not os.path.exists(path):
-                tty.die("No command for '{0}' was found.".format(name))
+            if '.' in name:
```

Comment:
Is there any reason not to always use the `else` branch? It would make the code a lot simpler.

Reply:
@adamjstewart I have so that the user can specify the suffix in case there are two files with the same name and different suffixes in the same directory. I don't really expect it to come up often, but I think it should be an option.

Are you suggesting removing that functionality, or refactoring it into a more similar codepath? I could make the only difference be adding `'.*'` to the name or not, but then we unnecessarily pass a static path to `glob.glob`.

Reply:
I'm saying that `foo.txt*` and `foo.txt` are most likely equivalent, so there's not really a reason to force `foo.txt*` to be `foo.txt`.

Reply:
I'm forcing `foo.txt` to be `foo.txt`, and `foo` to be `foo.*`

The `if` case is the former, and the `else` case the latter.

Reply:
Yes, I understand. But what I'm saying is that there almost certainly won't be any files like `food.txt.py`, so the `if` case isn't really necessary. Always adding the wildcard to the end should be fine, and would simplify the code.

Reply:
How about this. If there exists a file with this exact name, open it. If not, add a wildcard. That way we can handle the rare case where there are two files like:
```
foo.txt
foo.txt.py
```
and you want to open the former.

Reply:
The `if` case was designed to cover the case where they passed a full filename, but I'll check for that with existence instead.